### DISCUSSION TITLE
Conditions for event rule on script completion

### DISCUSSION THREAD

**Discussion by Azmodeszer** [netbox-community/netbox#17238] [Q&A]
I want to create an event rule that triggers email notifications if a script/job that runs daily ends in a specific state/contains a specific string. What I can't figure out so far is the syntax/logic for the condition and how to correctly reference the script instance.

I've selected "Extras | script" as Object types and checked "Events: Job terminations".

The JSON in the condition is
```
{
    "and": [
        {
            "attr": "id",
            "value": 13
        },
        {
            "attr": "result",
            "op": "contains",
            "value": "pending"
        }
    ]
}
```

With 13 being the ID of the script. To my most severe surprise this does not work. I'm not sure if I'm on the right track with referencing the script itself and not (somehow) the specific run instances?

Thanks in advance.

**Comment by Azmodeszer** [comment]
I guess I'm doing something wrong with the condition logic, because I'm also trying to get notifications for the creation of journal entries:

```
{
    "or": [
        {
            "attr": "kind",
            "value": "deletion"
        },
        {
            "attr": "kind",
            "value": "correction"
        }
    ]
}
```

If I leave the condition field blank, it works, of course, but if I want to restrict it to certain kinds... zilch.

**Comment by Azmodeszer** [comment]
And another thing... how do I pass JSON parameters to the email script?

The relevant portion of the script:
```
    def run(self, data, commit):
        subject = data.get('subject', 'NetBox Notification')
        body = data.get('body', 'Default Body')
        sender = data.get('sender', None)
        recipient = data.get('recipient', 'myemail@mail.com')

        send_mail(
            subject,
            body,
            sender,
            [recipient]
```

The action data in the Event Rule:
```
{
    "body": "Event rule-specific message body.",
    "subject": "Event rule-specific subject"
}

```
And yet the triggered email has subject "NetBox Notification" and body "Default Body".

**Comment by dung2k1deptrai** [comment]
> I want to create an event rule that triggers email notifications if a script/job that runs daily ends in a specific state/contains a specific string. What I can't figure out so far is the syntax/logic for the condition and how to correctly reference the script instance.
> 
> I've selected "Extras | script" as Object types and checked "Events: Job terminations".
> 
> The JSON in the condition is
> 
> ```
> {
>     "and": [
>         {
>             "attr": "id",
>             "value": 13
>         },
>         {
>             "attr": "result",
>             "op": "contains",
>             "value": "pending"
>         }
>     ]
> }
> ```
> 
> With 13 being the ID of the script. To my most severe surprise this does not work. I'm not sure if I'm on the right track with referencing the script itself and not (somehow) the specific run instances?
> 
> Thanks in advance.

I'm also experiencing this error; have you fixed it yet?

**Comment by Azmodeszer** [comment]
I wasn't able to figure it out, nope.
